internal/mining/cpuminer/cpuminer.go

```diff
@@ -826,14 +826,15 @@ out:
 //
 // See the documentation for CPUMiner type for more details.
 func New(cfg *Config) *CPUMiner {
-	return &CPUMiner{
+	miner := &CPUMiner{
 		g:                 cfg.BgBlkTmplGenerator,
 		cfg:               cfg,
-		numWorkers:        defaultNumWorkers,
 		updateNumWorkers:  make(chan struct{}),
 		queryHashesPerSec: make(chan float64),
 		speedStats:        make(map[uint64]*speedStats),
 		minedOnParents:    make(map[chainhash.Hash]uint8),
 		quit:              make(chan struct{}),
 	}
+	miner.numWorkers.Store(defaultNumWorkers)
```

Comment:
This change is probably worth extra scrutiny.

Previously the underlying `numWorkers` primitive was set directly **during** creation of the `CPUMiner`, but that is no longer possible so it is instead set using the proper atomic `Store` func **after** creation.

Reply:
this is fine. the pointer has not been returned to anyone else yet.

Reply:
Agreed.  There is no issue here as the pointer is not accessible to the caller yet as it hasn't been returned.